### ISSUE TITLE
Sails js websocket usage

### ISSUE THREAD

**Issue by angel1st** [balderdashy/sails#2775]
Hi there,
I am building WebRTC based application, using SailsJS on the server side. For the webrtc part I choose easyrtc full stack framework. easyrtc includes socket server as well. Since easyrtc requires initialization during the Sails lift, I have created hook for it.
Here is the code for easyrtc hook `initialize` method:

    module.exports = function easyrtcHook(sails) {
    'use strict';
    var io = require("socket.io");
    ......

    return {
        initialize: function (cb) {
            var eventsToWaitFor = ['hook:sockets:loaded', 'hook:http:loaded'];

            function listenCallback(err, pub) {
    
                 .....

            }
            
            sails.after(eventsToWaitFor, function () {
                // is it possible to use SailsJS socket, and if it is, 
                // what to pass instead of socketServer variable???  
                var socketServer = io.listen(sails.hooks.http, {"log level": 1}),
            
                // Start EasyRTC server with options to change the log level and add dates to the log.
                easyrtcServer = easyrtc.listen(sails.hooks.http.app,
                                                   socketServer,
                                                   {logLevel: "debug", logDateEnable: true},
                                                   listenCallback);

                // Then call cb()
                return cb();
            });
         }
      };
    };

As you can see, I successfully obtain access to `sails.hooks.http` - sails web server and pass it to `easyrtcServer`. I want to do the same with Sails socket, and my anticipation was that i have to pass `sails.hooks.sockets`, however I have got and error while attempting that. 
So my question is - is it possible to re-use SailsJS socket, instead of instantiate my own (as I did now), and if it is possible, what should I pass to `easyrtc.listen` instead of `socketServer` in my current code?

**Comment by angel1st** [comment]
OK, I have some progress there - I replaced `socketServer` with `sails.io`. Sails lifts successfully, however, easyrtc cannot establish the socket connection. My understanding is that Sails intercepts and does not send socket events emitted by EasyRTC.
So how can I deal with that issue?

**Comment by sailsbot** [comment]
Thanks for posting, @angel1st.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
